app/models/site_setting.rb

```diff
@@ -88,6 +88,12 @@ def self.anonymous_homepage
       .first
   end
 
+  def self.homepage_handler(request = nil, current_user = nil)
```

Comment:
Not sure the SiteSetting model is the best place for this, but it's useful to have a central place to check this since it's called from the application controller, application helper and the HomepageConstraint helper.

Reply:
How about naming it `resolved_homepage`? ('handler' made me think it was gonna return a method reference)

Reply:
Also I kinda agree SiteSetting isn't an ideal place. Feels very weird to be passing in `request` and `current_user` stuff here.

Maybe we make a dedicated `lib/homepage_resolver.rb` module for this logic?

Reply:
Opted for `HomepageHelper` as the name, with a `resolve()` method. That way, it's more open to future homepage-related helper methods.